src/vellum/workflows/nodes/tests/test_mocks.py

```diff
@@ -211,6 +211,96 @@ class StartNodeDisplay(BaseNodeDisplay[StartNode]):
     )
 
 
+def test_mocks__use_node_id_from_display():
+    """
+    Tests that validate_all correctly resolves mocks when the node ID is annotated by the display class.
+    """
+
+    # GIVEN a Base Node
+    class StartNode(BaseNode):
+        class Outputs(BaseNode.Outputs):
+            foo: str
+
+    # AND a display class that sets a custom node_id
+    display_node_id = uuid.uuid4()
+
+    class StartNodeDisplay(BaseNodeDisplay[StartNode]):
+        node_id = display_node_id
+
+    # AND a workflow class with that Node
+    class MyWorkflow(BaseWorkflow):
+        graph = StartNode
+
+        class Outputs(BaseWorkflow.Outputs):
+            final_value = StartNode.Outputs.foo
+
+    # AND a mock workflow node execution using the display-annotated node ID in legacy format
+    node_output_id = StartNode.__output_ids__["foo"]
+    raw_mock_workflow_node_execution = [
+        {
+            "type": "WORKFLOW_NODE_OUTPUT",
+            "node_id": str(display_node_id),
+            "mock_executions": [
+                {
```

Comment:
DevinAI This is the legacy shape, please use the `NODE_EXECUTION` shape. we should add this test to the `ee` directory as a result with the mock tests we have there